### PR TITLE
Feat state machine

### PR DESCRIPTION
**Pull request by sajelian**
created state machine for drone deployment

### CODE REVIEW

File: controllers/ppo/ppo_deploy.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class StateMachine:
+    """Class to handle the Drone State Machine transitions."""
+
+    def __init__(self, initial_goal: np.ndarray):
+        """Initialize the State Machine.
+
+        Args:
+            initial_goal: Stabilization goal the drone is trying to reach before landing.
+
```
Comment (danielsanjosepro) [inline controllers/ppo/ppo_deploy.py:29]: what 😆

File: controllers/ppo/ppo_deploy.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+from enum import Enum, auto  # noqa: D100
+from typing import Any, Dict
+
+import numpy as np
+
+from lsy_drone_racing.command import Command
+
+
+class DroneState(Enum):
```
Comment (danielsanjosepro) [inline controllers/ppo/ppo_deploy.py:9]: first time that I see the auto function in action with enums, cool

File: controllers/ppo/ppo.py [hunk 3]
```diff
@@ -23,46 +24,33 @@ def __init__(
...
-        Args:
-            initial_obs: The initial observation of the environment's state. Consists of
-                [drone_xyz_yaw, gates_xyz_yaw, gates_in_range, obstacles_xyz, obstacles_in_range,
-                gate_id]
-            initial_info: The a priori information as a dictionary with keys 'symbolic_model',
-                'nominal_physical_parameters', 'nominal_gates_pos_and_type', etc.
-            buffer_size: Size of the data buffers used in method `learn()`.
-            verbose: Turn on and off additional printouts and plots.
-        """
+        """Initialization of the controller."""
```
Comment (danielsanjosepro) [inline controllers/ppo/ppo.py:27]: why did you remove the docs for the function?

File: controllers/ppo/ppo.py [hunk 4]
```diff
@@ -71,41 +59,21 @@ def compute_control(
...
-        target_yaw = float(action[3])
-        target_rpy_rates = np.zeros(3)
-        args = [target_pos, target_vel, target_acc, target_yaw, target_rpy_rates, ep_time]
-
-        # logger.info(f"Predicted action: {action}")
-        # logger.info(f"Args: {args}")
-        return command_type, args
-
+        """Pick command sent to the quadrotor through a Crazyswarm/Crazyradio-like interface."""
+        if ep_time - 2 > 0 and info["current_gate_id"] != -1:
```
Comment (danielsanjosepro) [inline controllers/ppo/ppo.py:63]: shouldnt it here use the information of the state machine something like `if self.state_machine.state == DroneState.PolicyControl: <then do policy control>`

File: controllers/ppo/ppo_deploy.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    TAKEOFF = auto()
+    POLICY_CONTROL = auto()
+    NOTIFY_SETPOINT_STOP = auto()
+    GOTO = auto()
+    LAND = auto()
+    FINISHED = auto()
+    NONE = auto()
+
+
+class StateMachine:
```
Comment (danielsanjosepro) [inline controllers/ppo/ppo_deploy.py:21]: Just as a preference, I prefer a more specific name like DroneStateMachine or similar so that it is cleaar that StateMachine is not a general StateMachine(like a Base class) but just specific to this problem. This is only a preference

**Review by danielsanjosepro — COMMENTED** [review]
all in all good I like the solution with the state machine

File: controllers/ppo/ppo_deploy.py (hunk 1 above)
Comment (sajelian) [reply controllers/ppo/ppo_deploy.py:29]: :laughing:

File: controllers/ppo/ppo.py (hunk 4 above)
Comment (sajelian) [reply controllers/ppo/ppo.py]: moved logic to deploy

**Review by danielsanjosepro — APPROVED** [review]
LGTM keep up the good work. You are amazing
